engine/runtime/src/main/java/org/enso/interpreter/runtime/EnsoContext.java

```diff
@@ -80,7 +80,7 @@ public class EnsoContext {
   private final AtomicLong clock = new AtomicLong();
 
   private final Shape rootStateShape = Shape.newBuilder().layout(State.Container.class).build();
-  private final ExecutionEnvironment executionEnvironment;
+  private ExecutionEnvironment executionEnvironment;
```

Comment:
Is this the only needed change to be able to modify `executionEnvironment`? Interesting, that's way simpler than I expected.

The only problem is that now the `executionEnvironment` isn't going to be a compile time constant - even when running in CLI mode - e.g. the check will always have to happen.

We should:
- make it `@CompilationFinal`
- shield it with `Assumption`
- that shall instruct the Truffle Graal compiler to treat it as a constant - the code will be simpler/faster
- if it changes for the first time, invalidate the assumption
- then read the field as non-final - e.g. then the code is going to be slower but flexible